kernel/data-pipeline/core/src/main/java/org/apache/shardingsphere/data/pipeline/core/check/consistency/algorithm/AbstractStreamingDataConsistencyCalculateAlgorithm.java

```diff
@@ -78,13 +78,13 @@ public boolean hasNext() {
         @Override
         public DataConsistencyCalculatedResult next() {
             calculateIfNecessary();
-            Optional<DataConsistencyCalculatedResult> nextResult = this.nextResult.get();
+            Optional<DataConsistencyCalculatedResult> nextCalculatedResult = this.nextResult.get();
             this.nextResult.set(null);
-            return nextResult.orElseThrow(NoSuchElementException::new);
+            return nextCalculatedResult.orElseThrow(NoSuchElementException::new);
         }
         
         private void calculateIfNecessary() {
-            if (null == nextResult.get()) {
+            if (!nextResult.get().isPresent()) {
                 nextResult.set(calculateChunk(param));
             }
```

Comment:
There's `this.nextResult.set(null);`, so `nextResult.get()` might be null.

`if (null == nextResult.get())` looks strange, since `calculateChunk` return `Optional`, there's no better way for now.

Reply:
change reverted